src/lib.rs

```diff
@@ -194,7 +196,10 @@ pub fn encode_upper<T: AsRef<[u8]>>(data: T) -> String {
 /// assert!(hex::decode("foo").is_err());
 /// ```
 pub fn decode<T: AsRef<[u8]>>(data: T) -> Result<Vec<u8>, FromHexError> {
-    FromHex::from_hex(data)
+    let mut result = vec![0; data.as_ref().len() / 2];
+    decode_to_slice(data, &mut result)?;
+
+    Ok(result)
```

Comment:
Why replace one line of code with three more complex ones? Are there performance advantages? Otherwise I'd prefer the single line.